traffic_ops/traffic_ops_golang/routes.go

```diff
@@ -407,6 +408,9 @@ func Routes(d ServerData) ([]Route, []RawRoute, http.Handler, error) {
 		{1.1, http.MethodGet, `deliveryservices/{id}/urlkeys/?(\.json)?$`, deliveryservice.GetURLKeysByID, auth.PrivLevelReadOnly, Authenticated, nil},
 		{1.1, http.MethodGet, `riak/bucket/{bucket}/key/{key}/values/?(\.json)?$`, apiriak.GetBucketKey, auth.PrivLevelAdmin, Authenticated, nil},
 
+		//Pattern based consistent hashing endpoint
+		{1.5, http.MethodPost, `consistenthash/?$`, consistenthash.Post, auth.PrivLevelReadOnly, Authenticated, nil},
```

Comment:
Can you add this endpoint to roles/capabilities so it is not lost when we migrate to roles/capabilities?

Off the top of my head probably do this:
1. create a capability called consistenthash-read or consistenthash-generate
2. add the capability to the admin, ops and r/o role
3. add the POST api/consistenthash endpoint to the consistenthash-read capability

you can follow the examples in seeds.sql